packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled ("ON").
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label of the Switch component.
+   */
+  label: PropTypes.node.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Parameters: 1. event 2. switch value
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  id,
+  labelId,
+  checked,
+  disabled,
+  onChange,
+  label,
+  ...customProps
+}) => {
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    sliderButtonRef.current.focus();
```

Comment:
We'll want to conditionally focus the switch control or let the native focus handle it and remove any custom programatic focus. The button should be able to handle browser native focus implementations. I added console logs to onBlur and onFocus and they are called on every switch update.

```jsx
      <Switch
        id="defaultSwitchOn"
        labelId="label3"
        checked={value}
        label="Label"
        onFocus={() => console.log('Focus')}
        onBlur={() => console.log('Blur')}
        onChange={(event, checked) => { setValue(!checked); }}
      />
```

Reply:
custom programatic focus was added for the focus issue that happens on Safari, FireFox and IE. In order to get focus to the button on click in firefox and safari we need to set focus manually to button on button click. 

The reason onBlur and onFocus being called on every switch change is because we are clicking on label. whenever we click on label onBlur of button gets triggered and on switch change button gets focus back. this is due to default behavior of htmlFor ( Label button association ) which triggers button onClick even when click was done on label.

I Observe similar behaviour with other form-controls like checkbox and radio-button that on clicking of label will trigger onBlur of checkbox before onClick.

I am not sure on the fix  for this issue, but I think adding onClick handlers on label instead of button to prevent onBlur from getting triggered might resolve this. but would like to know if there are any other ways to fix this issue.

Reply:
the repeated callbacks issue has been fixed by handling [mousedown](https://github.com/cerner/terra-core/blob/48fce2d9ca73773ac6f9891feaa7f2a56b36f864/packages/terra-switch/src/Switch.jsx#L103) event [here](https://github.com/cerner/terra-core/blob/48fce2d9ca73773ac6f9891feaa7f2a56b36f864/packages/terra-switch/src/_SwitchUtils.js#L15)